src/execution_plan/ops/op_project.c

```diff
@@ -75,14 +130,17 @@ static Record ProjectConsume(OpBase *opBase) {
 		 * MATCH (a) WITH toUpper(a.name) AS e RETURN e
 		 * TODO This is a rare case; the logic of when to persist can be improved.  */
 		if(!(v.type & SI_GRAPHENTITY)) SIValue_Persist(&v);
+		if(op->intermediate_record)
+			Record_Add(op->intermediate_record, op->intermediate_record_ids[i], v);
 		Record_Add(projection, rec_idx, v);
 	}
 
-	uint output_exp_count = array_len(op->output_exps);
-	for(uint i = 0; i < output_exp_count; i++) {
-		AR_ExpNode *exp = op->output_exps[i];
-		SIValue v = AR_EXP_Evaluate(exp, projection);
-		int rec_idx = op->record_offsets[input_exp_count + i];
+	// Evaluate expressions on the intermediate Record, if any.
+	uint order_count = array_len(op->order_exps);
```

Comment:
Consider saving both `order_count` and `projection_count` within `OpProject` just so we can save all of the calls to `array_len`